src/subprocess/runner.ts

```diff
@@ -213,39 +183,56 @@ class Reporter extends reporters.Base {
       event: 'sync',
     });
 
+    assert(event);
+
     // and then send the event
     process.send!({ event, data });
   }
 }
 
-const mocha = new Mocha();
-mocha.addFile(argv.test);
+process.on('message', (msg) => {
+  try {
+    msg = JSON.parse(msg);
+  } catch (_) {
+    return;
+  }
+  const { type, testOptions } = msg;
 
-// --compilers
-applyCompilers(argv.compilers);
+  if (type !== 'start') {
+    return;
+  }
 
-// --delay
-applyDelay(mocha, argv.delay);
+  const mocha = new Mocha();
+  mocha.addFile(testOptions.test);
 
-// --grep
-applyGrepPattern(mocha, argv.grep);
+  // --compilers
+  applyCompilers(testOptions.compilers);
 
-// --enableTimeouts
-applyNoTimeouts(mocha, argv.enableTimeouts);
+  // --delay
+  applyDelay(mocha, testOptions.delay);
 
-// --exit
-const onComplete = applyExit(argv.exit);
+  // --grep
+  applyGrepPattern(mocha, testOptions.grep);
 
-// --require
-applyRequires(argv.require);
+  // --enableTimeouts
+  applyNoTimeouts(mocha, testOptions.enableTimeouts);
 
-// --timeout
-applyTimeouts(mocha, argv.timeout);
+  // --exit
+  const onComplete = applyExit(testOptions.exit);
 
-// apply main process root suite properties
-for (const option of SUITE_OWN_OPTIONS) {
-  const suiteProp = `_${option}`;
-  mocha.suite[suiteProp] = argv[option];
-}
+  // --require
+  applyRequires(testOptions.require || []);
+
+  // --timeout
+  applyTimeouts(mocha, testOptions.timeout);
+
+  // apply main process root suite properties
+  for (const option of SUITE_OWN_OPTIONS) {
+    const suiteProp = `_${option}`;
+    mocha.suite[suiteProp] = testOptions[option];
+  }
 
-mocha.reporter(Reporter).run(onComplete);
+  mocha.reporter(Reporter).run((code) => {
```

Comment:
You can just write `mocha.reporter(Reporter).run(onComplete)` I guess?

Reply:
Fixed